### ISSUE TITLE
regexp cause cronicle daemon can not start，how to solve it?

### ISSUE THREAD

**Issue by luojieio** [jhuckaby/Cronicle#88]
SyntaxError: Invalid regular expression: /172-31-0-9|[^172-31-0-2$]|172-31-0-1[0234589]|172-31-0-2[02358]|172-31-0-3[034567]|172-31-0-4|172-31-0-4[1]|172-31-0-6|172-31-0-4[03]/: Range out of order in character class
    at new RegExp (<anonymous>)
    at /data/cronicle/Cronicle-0.8.16/lib/engine.js:186:19
    at /data/cronicle/Cronicle-0.8.16/node_modules/pixl-server-storage/list.js:535:12
    at /data/cronicle/Cronicle-0.8.16/node_modules/async/dist/async.js:3874:9
    at /data/cronicle/Cronicle-0.8.16/node_modules/async/dist/async.js:473:16
    at replenish (/data/cronicle/Cronicle-0.8.16/node_modules/async/dist/async.js:993:25)
    at iterateeCallback (/data/cronicle/Cronicle-0.8.16/node_modules/async/dist/async.js:983:17)
    at /data/cronicle/Cronicle-0.8.16/node_modules/async/dist/async.js:958:16
    at /data/cronicle/Cronicle-0.8.16/node_modules/async/dist/async.js:3871:13
    at /data/cronicle/Cronicle-0.8.16/node_modules/pixl-server-storage/list.js:495:6

**Comment by jhuckaby** [comment]
Thank you for this bug report.  It looks like you've entered a malformed regular expression as a group server match.  However, it isn't properly validated, so the server crashes trying to run it.

This should be fixed in [Cronicle v0.8.18](https://github.com/jhuckaby/Cronicle/releases/tag/v0.8.18).  However, you may have to manually fix the value in your server group data.  Is this regular expression is part of your **master** server group?

```
/172-31-0-9|[^172-31-0-2$]|172-31-0-1[0234589]|172-31-0-2[02358]|172-31-0-3[034567]|172-31-0-4|172-31-0-4[1]|172-31-0-6|172-31-0-4[03]/
```

If so, then you will have to perform a manual edit to fix it.  On your master server, you will have to edit a JSON data record and correct the regular expression syntax.  Enter this command to spawn your local terminal editor (`vi` or other):

```
/opt/cronicle/bin/control.sh stop
/opt/cronicle/bin/storage-cli.js edit global/server_groups/0
```

Then carefully correct the malformed regular expression, save and exit your text editor, then restart Cronicle:

```
/opt/cronicle/bin/control.sh start
```

**Comment by luojieio** [comment]
Thanks
